app/models/user.rb

```diff
@@ -24,16 +24,9 @@ class User < ApplicationRecord
   def self.find_for_database_authentication(warden_conditions)
     conditions = warden_conditions.dup
     if login = conditions.delete(:login)
-      where(conditions.to_h).where(["lower(username) = :value OR lower(email) = :value", { :value => login.downcase }]).first
-    elsif conditions.has_key?(:username) || conditions.has_key?(:email)
-      conditions[:email].downcase! if conditions[:email]
-      conditions[:username].downcase! if conditions[:username]
-      where(conditions.to_h).first
+      return where(conditions.to_h).where(["username = :value OR lower(email) = :value", { :value => login.downcase }]).first
     end
+    super(warden_conditions)
```

Comment:
So, I tested this (by inserting a raise and debugging from there) and it's safer to use `conditions` than it is to use `warden_conditions` because there is no `login` column on the users table.

So I was able to confirm that there was a parent method, and that it required conditions that correlate to the user table.